includes/api/validation-functions.php

```diff
@@ -331,6 +368,29 @@ function validate_email_field( $value, array $field ): void {
 	}
 }
 
+/**
+ * Validate for valid number type.
+ *
+ * @param mixed  $value The value.
+ * @param mixed  $number_type The number type.
+ * @param string $message Optional. The error message.
+ *
+ * @throws Validation_Exception Exception when value is invalid.
+ *
+ * @return void
+ */
+function validate_number_type( $value, $number_type, string $message = '' ): void {
+	$message = $message ?: \__( 'Value must be a valid number', 'atlas-content-modeler' );
+
+	if ( $number_type === 'integer' && ! is_int( $value ) ) {
```

Comment:
I think this is why the tests are failing. You might have to check for integers in a string-safe way, because WP stores field data as strings:

```suggestion
	if ( $number_type === 'integer' && ! filter_var( $value, FILTER_VALIDATE_INT ) ) {
```

Reply:
@nickcernis is on the right path here. Since this could be submitted from a form, the value could be a string.

`filter_var` seems to work the best here. Although it does return `1` for `true`.

```
wp> filter_var( "1", FILTER_VALIDATE_INT )
=> int(1)
wp> filter_var( "1.2", FILTER_VALIDATE_INT )
=> bool(false)
wp> filter_var( "a", FILTER_VALIDATE_INT )
=> bool(false)
wp> filter_var( true, FILTER_VALIDATE_INT )
=> int(1)
```

But if `validate_number()` is ran before this, it should catch the possible `true` value.